### PR TITLE
Stop testing on FreeBSD 12

### PR DESCRIPTION
**Pull request by asomers**
FreeBSD 12.4 will be EoL at the end of the month.  Stop testing on it.  Also, update the 14.0 image to the official release.

### CODE REVIEW

**Comment by devnexen** [comment]
There is a [PR for it](https://github.com/rust-lang/libc/pull/3474).

File: .cirrus.yml [hunk 1]
```diff
@@ -29,7 +15,7 @@ task:
 task:
   name: nightly x86_64-unknown-freebsd-14
   freebsd_instance:
-    image: freebsd-14-0-rc1-amd64
+    image: freebsd-14-0-release-amd64-ufs
```
Comment (devnexen) [inline .cirrus.yml:18]: now there is a proper 14 release, `image_family: freebsd-14-0` is sufficient.

File: .cirrus.yml [hunk 2]
```diff
@@ -1,17 +1,3 @@
-task:
```
Comment (devnexen) [inline .cirrus.yml:1]: it s tricky ... logically you re right but in the wild there are many users still taking time to migrate (in another project I had a freebsd ... 9.2 user, a bit extreme :) but you get me ..). I may suggest to give a bit of time before removing it but not up to me..

Comment (asomers) [reply .cirrus.yml:1]: 9.2?  Wow.  My feeling is that while some users may take some time to upgrade, they also probably aren't using cutting edge Rust software.  So those users can probably make do with older crates.  No need to maintain the latest Rust releases for old FreeBSD release.  At least, that's my philosophy.

Comment (devnexen) [reply .cirrus.yml:1]: Fair enough. It seems it takes more than this tough see [what it took for freebsd 11](https://github.com/rust-lang/homu/pull/169).

**Comment by JohnTitor** [comment]
r+ once https://github.com/rust-lang/homu/pull/199 gets merged.

File: .cirrus.yml (hunk 2 above)
Comment (JohnTitor) [reply .cirrus.yml:1]: Note that removing CI doesn't mean that the existing usage on FreeBSD 12 will be broken. The impact of this should be low, I think.

**Review by JohnTitor — APPROVED** [review]

**Pull request merged**